specifyweb/frontend/js_src/lib/components/QueryBuilder/Results.tsx

```diff
@@ -99,6 +100,20 @@ export function QueryResults(props: QueryResultsProps): JSX.Element {
     canFetchMore,
   } = useFetchQueryResults(props);
 
+  const isEmbeddedCollectingEvent = schema.embeddedCollectingEvent;
+  const isEmbeddedPaleoContext = schema.embeddedPaleoContext;
+  const canMerge =
+    hasPermission('/record/merge', 'update') &&
+    hasTablePermission(table.name, 'update');
+  const canMergePaleoContext =
+    table.name === 'PaleoContext' && !isEmbeddedPaleoContext && canMerge;
+  const canMergeCollectingEvent =
+    table.name === 'CollectingEvent' && !isEmbeddedCollectingEvent && canMerge;
+  const canMergeOtherTables =
+    table.name !== 'PaleoContext' &&
+    table.name !== 'CollectingEvent' &&
+    canMerge;
```

Comment:
could you move this logic into a function? you will give that function the table and it will return a boolean - based on the value of that boolean you will render or not RecordMergingLink